core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -110,6 +112,7 @@ final class HttpChannelPool implements AsyncCloseable {
                                           SessionProtocol.H2, SessionProtocol.H2C));
         pendingAcquisitions = newEnumMap(httpAndHttpsValues());
         allChannels = new IdentityHashMap<>();
+        connectionEventStates = new IdentityHashMap<>();
```

Comment:
Question) What is the reason we need this map?

Reply:
When the sessionPromise failed, I could not get the channel because the future.getNow() returns null. 
Also, I couldn't pass the `ConnectionEventState` to the `notifyConnectionFailed` method, because the `ConnectionEventState` is created in the callback.